test/oli_web/live/admin/institutions/index_live_test.exs

```diff
@@ -0,0 +1,266 @@
+        insert(:pending_registration, %{
+          name: "My new institution"
+        })
+
+      assert Institution
+             |> where([i], i.name == "My new institution")
+             |> Repo.one() == nil
+
+      {:ok, view, _html} = institutions_index_route(conn)
+      go_to_pending_registrations(view)
+
+      view
+      |> render_click("select_pending_registration", %{
+        registration_id: pending_registration.id,
+        action: "review"
+      })
+
+      view
+      |> render_submit(
+        "save_registration",
+        %{
+          registration:
+            pending_registration
+            |> Map.from_struct()
+            |> Map.take([
+              :country_code,
+              :institution_email,
+              :institution_url,
+              :name,
+              :issuer,
+              :client_id,
+              :deployment_id,
+              :key_set_url,
+              :auth_token_url,
+              :auth_login_url,
+              :auth_server,
+              :line_items_service_domain
+            ])
+        }
+      )
+
+      assert Institution
+             |> where([i], i.name == "My new institution")
+             |> Repo.one() != nil
+    end
+
+    test "when a pending registration contains the url of an existing institution, the existing institution is used instead",
+         %{conn: conn} do
+      existing_institution =
+        insert(:institution, %{
+          institution_url: "www.existing_institution.com",
+          name: "Existing institution"
+        })
+
+      pending_registration =
+        insert(:pending_registration, %{
+          institution_url: "www.existing_institution.com",
+          name: "Existing institution with different name"
+        })
+
+      {:ok, view, _html} = institutions_index_route(conn)
+      go_to_pending_registrations(view)
+
+      view
+      |> render_click("select_pending_registration", %{
+        registration_id: pending_registration.id,
+        action: "review"
+      })
+
+      assert has_element?(
+               view,
+               "select[name=\"registration[institution_id]\"] option[value=\"#{existing_institution.id}\"][selected=\"selected\"]"
+             )
+
+      view
+      |> render_submit(
+        "save_registration",
+        %{
+          registration:
+            pending_registration
+            |> Map.from_struct()
+            |> Map.take([
+              :country_code,
+              :institution_email,
+              :institution_url,
+              :name,
+              :issuer,
+              :client_id,
+              :deployment_id,
+              :key_set_url,
+              :auth_token_url,
+              :auth_login_url,
+              :auth_server,
+              :line_items_service_domain
+            ])
+        }
+      )
+
+      assert Institution
+             |> where([i], i.name == "Existing institution with different name")
+             |> Repo.one() == nil
+    end
+  end
+
```

Comment:
Should we test that we are allowed to add a new institution when reviewing a registration?

Reply:
That's actually being tested in `test/oli_web/live/admin/institutions/index_live_test.exs:147`. If a pending registration doesn't match any of the existing institution urls, a new institution gets created